indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentAllocationQueue.java

```diff
@@ -196,6 +204,7 @@ public Future<SegmentIdWithShardSpec> add(SegmentAllocateRequest request)
       }
     });
 
+    scheduleQueuePoll(maxWaitTimeMillis);
```

Comment:
Should this be part of the method `addBatchToQueue` itself?
1. With the current code, I think it's possible that requeued batches may not be processed if new requests don't come in.
2. I also don't think we need to schedule a poll for every new request, but just for the first one for a batch.

Reply:
Yeah, makes sense. Let me fix it up.

Reply:
Fixed.